File: ChessAPI/Domain/Entities/Player.cs
using Domain.Enums;

namespace Domain.Entities;

public class Player
{
    public int Id { get; set; }
    public string Pseudo { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string PasswordHash { get; set; } = string.Empty;
    public DateTime BirthDate { get; set; }
    public Genre Genre { get; set; }
    public int Elo { get; set; } = 1200;
}

File: ChessClient/src/app/models/player.ts
export enum Genre {
    Male = 0,
    Female = 1,
    NonBinary = 2,
    PreferNotToSay = 3
}

export interface Player {
  id: number;
  pseudo: string;
  email: string;
  birthDate: string;
  genre: Genre;
  elo: number;
}
